core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -447,16 +447,9 @@ class ControllerApis(val requestChannel: RequestChannel,
         requestThrottleMs => apiVersionRequest.getErrorResponse(requestThrottleMs, INVALID_REQUEST.exception))
       CompletableFuture.completedFuture[Unit](())
     } else {
-      val context = new ControllerRequestContext(request.context.header.data, request.context.principal, OptionalLong.empty())
-      controller.finalizedFeatures(context).handle { (result, exception) =>
-        requestHelper.sendResponseMaybeThrottle(request, requestThrottleMs => {
-          if (exception != null) {
-            apiVersionRequest.getErrorResponse(requestThrottleMs, exception)
-          } else {
-            apiVersionManager.apiVersionResponse(requestThrottleMs, result.featureMap().asScala.toMap, result.epoch())
-          }
-        })
-      }
+      requestHelper.sendResponseMaybeThrottle(request,
```

Comment:
It cannot be the same as KafkaApis because this code returns a future, which the KafkApis code does not. And the comment about authentication seems good to keep.